src/loaders/api/api.js

```diff
@@ -26,15 +28,20 @@ function setTopLevelCallers (nr) {
   })
 
   function caller (fnName, ...args) {
+    let returnVals = []
     Object.values(nr.initializedAgents).forEach(val => {
-      if (val.exposed && val.api[fnName]) val.api[fnName](...args)
+      if (val.exposed && val.api[fnName]) {
+        returnVals.push(val.api[fnName](...args))
+      }
     })
+    return returnVals.length > 1 ? returnsVals : returnVals[0]
   }
 }
 
-export function setAPI (agentIdentifier, nr, forceDrain) {
+export function setAPI (agentIdentifier, forceDrain) {
   if (!forceDrain) registerDrain(agentIdentifier, 'api')
-  setTopLevelCallers(nr)
+  const apiInterface = {}
+  setTopLevelCallers()
```

Comment:
`setTopLevelCallers` is global scope while `setAPI` is agent scope. Can they be separated, and have _configure.js_ make the call to `setTopLevelCallers`?

Reply:
Yes i think that sounds reasonable

Reply:
fixed